File: src/main/java/resonantinduction/archaic/crate/RenderCrate.java
package resonantinduction.archaic.crate;

import net.minecraft.tileentity.TileEntity;
import resonantinduction.core.render.RenderItemOverlayTile;
import cpw.mods.fml.relauncher.Side;
import cpw.mods.fml.relauncher.SideOnly;

@SideOnly(Side.CLIENT)
public class RenderCrate extends RenderItemOverlayTile
{
	@Override
	public void renderTileEntityAt(TileEntity tileEntity, double x, double y, double z, float var8)
	{
		if (tileEntity instanceof TileCrate)
		{
			TileCrate tile = (TileCrate) tileEntity;
			renderItemOnSides(tileEntity, tile.getSampleStack(), x, y, z, "Empty");
		}
	}
}

File: src/main/scala/resonantinduction/archaic/crate/RenderCrate.scala
package resonantinduction.archaic.crate

import net.minecraft.client.renderer.tileentity.TileEntitySpecialRenderer
import net.minecraft.tileentity.TileEntity
import org.lwjgl.opengl.GL11
import resonant.lib.render.RenderItemOverlayUtility
import resonant.lib.utility.LanguageUtility
import cpw.mods.fml.relauncher.Side
import cpw.mods.fml.relauncher.SideOnly

@SideOnly(Side.CLIENT) class RenderCrate extends TileEntitySpecialRenderer {
  def renderTileEntityAt(tileEntity: TileEntity, x: Double, y: Double, z: Double, var8: Float)
  {
    if (tileEntity.isInstanceOf[TileCrate])
    {
      GL11.glPushMatrix
      val tile: TileCrate = tileEntity.asInstanceOf[TileCrate]
      RenderItemOverlayUtility.renderItemOnSides(tileEntity, tile.getSampleStack, x, y, z, LanguageUtility.getLocal("tooltip.empty"))
      GL11.glPopMatrix
    }
  }
}